elastic/beats
# Fix logstash loadavg types (non-xpack cases)

## What does this PR do?

Switches logtash node loadavg from long to half_float.

## Why is it important?

Load average is a floating point value. Mapping as long causes the graph to display 0, 1, 2 which is not great granularly for loadaverages.

This change will provide enough precision to get a nice smooth load graph like this one.

![Screen Shot 2022-09-29 at 15 15 29](https://user-images.githubusercontent.com/690/192953430-2dc460ed-5820-4780-9640-88f61cc524cb.png)

## Checklist




## Author's Checklist



## How to test this PR locally

Same as steps in https://github.com/elastic/elasticsearch/pull/90494 but run with `xpack.enabled: false` and confirm the non-integer values in the `metricbeat-*` indices manually.

## Related issues

Discovered while working on https://github.com/elastic/kibana/issues/142179

Closes https://github.com/elastic/kibana/issues/142181

https://github.com/elastic/elasticsearch/pull/90494 contains the corresponding change for the mapping used when running in xpack mode to drive the stack monitoring UI.

## Use cases



## Screenshots



## Logs
